tests/test_openapi.py

```diff
@@ -45,3 +56,465 @@ def test_str2bool():
+    return 'tests/{}'.format(filename)
+
+
+@pytest.fixture()
+def config():
+    """
+    Get the OpenAPI Document configuration
+
+    :returns: OpenAPI configuration YAML dict
+    """
+    with open(get_test_file_path('pygeoapi-test-openapi-config.yml')) as config_file: # noqa
+        return yaml_load(config_file)
+
+
+@pytest.fixture()
+def get_oas_30_(config):
+    """
+    Get the generated OpenAPI 3.0 Document
+
+    :param config: configuration object
+
+    :returns: OpenAPI definition YAML dict
+    """
+    return get_oas_30(config)
+
+
+@pytest.fixture()
+def is_cql(config):
+    """
+    Checks whether any resource supports feature filter
+
+    :param config: configuration object
+
+    :returns: Boolean value
+    """
+    collections = filter_dict_by_key_value(config['resources'],
+                                           'type', 'collection')
+    for k, _ in collections.items():
+        if 'filters' in collections[k]:
+            return True
+    return False
+
+
+@pytest.fixture()
+def get_cql_components(get_oas_30_):
+    """
+    Get the generated OpenAPI 3.0 Document
+
+    :param get_oas_30_: OpenAPI 3.0 Document object
+
+    :returns: OpenAPI Document components YAML dict
+    """
+    openapi_components = get_oas_30_.get('components', None)
+    return openapi_components
+
+
+@pytest.fixture()
+def get_cql_schemas(get_cql_components):
+    """
+    Get the schemas from OpenAPI 3.0 Document
+
+    :param get_cql_components: OpenAPI 3.0 Document components object
+
+    :returns: OpenAPI Document schemas YAML dict
+    """
+    openapi_schemas = get_cql_components.get('schemas', None)
+    return openapi_schemas
+
+
+def test_cql_paths(config, get_oas_30_, get_cql_components, is_cql):
+    """
+    Assertions for CQL paths in OpenAPI 3.0 Document
+
+    :param config: configuration object
+    :param get_oas_30_: OpenAPI 3.0 Document object
+    :param get_cql_components: OpenAPI 3.0 Document components
+    :param is_cql: boolean value
+    """
+    assert isinstance(config, dict)
+    assert isinstance(get_oas_30_, dict)
+
+    if is_cql:
+
+        cql_paths = get_oas_30_.get('paths', None)
+        # assertion for get paths
+        assert cql_paths is not None
+
+        collections = filter_dict_by_key_value(config['resources'],
+                                               'type', 'collection')
+        for k, _ in collections.items():
+            references = []
+            for items_parameters in cql_paths['/collections/'+k+'/items']['get']['parameters']: # noqa
+                if '$ref' in items_parameters:
+                    references.append(items_parameters['$ref'])
+            # if the resource support filter
+            if 'filters' in collections[k]:
+                assert '#/components/parameters/filter' in references # noqa
+                assert '#/components/parameters/filter-lang' in references  # noqa
+            # if the resource does not support filter
+            else:
```

Comment:
In reference to https://github.com/geopython/pygeoapi/pull/478#pullrequestreview-443577941
@jorgejesus @francbartoli assertion when resources does not support filter